src/internal/SignalGenerator.cpp

```diff
@@ -0,0 +1,36 @@
+#include <bits/stdc++.h>
+#include <hyro/SignalGenerator.h>
+
+namespace hyro 
+{
+
+SignalGenerator::SignalGenerator()
+{
+  m_deg = 0.0;
+  m_step = 1.0;
+  m_amp = 1.0; 
+}
+
+double SignalGenerator::getSignalValue()
+{
+  rad = m_deg / 180.0 * M_PI;
+  m_deg += m_step;
+  
+  if (cosine) 
+    return m_amp * cos(rad);
+  return  m_amp * sin(rad);
+}
+
+void SignalGenerator::setAmplitude(double amplitude) { m_amp = amplitude; }
```

Comment:
Please, do not use this style for functions in the CPP files. You can move the getters and setters functions to the header, and the you can declare the functions using one line.

In header file you can do:

```c++
// In the .h file
class SignalGenerator
{
 void setAmplitude(double amplitude) { m_amp = amplitude; }
};
```

However, in the CPP or implementation file, you never declare a function using one line.

```c++
// In the .cpp file
void SignalGenerator::setAmplitude(double amplitude)
{ 
  m_amp = amplitude;
}
```


void SignalGenerator::setAmplitude(double amplitude) { m_amp = amplitude; }